service/pkg/auth/context_auth.go

```diff
@@ -0,0 +1,64 @@
+package auth
+
+import (
+	"context"
+
+	"github.com/lestrrat-go/jwx/v2/jwk"
+	"github.com/lestrrat-go/jwx/v2/jwt"
+	"github.com/opentdf/platform/service/logger"
+)
+
+const (
+	authnContextKey = authContextKey("dpop-jwk")
+)
+
+type authContextKey string
```

Comment:
Can we take this time to update the context key to a struct. Its better practice and reduces the risk of this key getting overridden.

`type authContextKey struct{}`

Reply:
i think the only advantage of using a struct over a custom typed string it that youre not allocating as much memory. "Users of WithValue should define their own types for keys. To avoid allocating when assigning to an interface{}, context keys often have concrete type struct{}" https://pkg.go.dev/context#WithValue
wouldnt authContextKey("dpop-jwk") and authContextKey{"key":""dpop-jwk"} have a similar risk of getting overrriden

Reply:
updated to use struct